applications/rag/README.md

```diff
@@ -1,192 +1,204 @@
 
-## Installation
+#### RAG on GKE Architecture
+1. A GKE service endpoint serving [Hugging Face TGI inference](https://huggingface.co/docs/text-generation-inference/en/index) using `mistral-7b`.
+2. [Cloud SQL `pgvector` instance](https://github.com/pgvector/pgvector) with vector embeddings generated from an input dataset.
+3. A [Ray](https://docs.ray.io/en/latest/ray-overview/getting-started.html) cluster running on GKE that runs jobs to generate embeddings and populate the vector DB.
+5. A [Jupyter](https://docs.jupyter.org/en/latest/) notebook running on GKE that reads the dataset using GCS fuse driver integrations and runs a Ray job to populate the vector DB.
+3. A front end chat interface running on GKE that prompts the inference server with context from the vector DB.
+
+The tutorial walks you through installing the RAG infrastructure in a GCP project, generating vector embeddings for a sample [Kaggle Netflix shows](https://www.kaggle.com/datasets/shivamb/netflix-shows) dataset and prompting the LLM with context.
+
+## Prerequisites
 
-Preinstall the following on your computer:
-* Kubectl
-* Terraform
-* Helm
-* Gcloud
+### (Required) Install tooling
 
-**NOTE:** Terraform keeps state metadata in a local file called `terraform.tfstate`. Deleting the file may cause some resources to not be cleaned up correctly even if you delete the cluster. We suggest using `terraform destory` before reapplying/reinstalling.
+Install the following on your computer:
+* [Kubectl](https://kubernetes.io/docs/tasks/tools/#kubectl)
+* [Terraform](https://developer.hashicorp.com/terraform/tutorials/aws-get-started/install-cli)
+* [Helm](https://helm.sh/docs/intro/install/)
+* [Gcloud](https://cloud.google.com/sdk/docs/install)
 
 ### (Optional) Use an existing cluster
 
-By default, Terraform creates an Autopilot cluster on your behalf. If you prefer to manage your own cluster, you will need to set `create_cluster = false` in the following section. Creating a long-running cluster is also better for development, allowing you to iterate on Terraform components without recreating the cluster every time.
+By default, this tutorial creates an Autopilot cluster on your behalf. We highly recommend following the default settings.
 
-You can use gcloud to create a GKE Autopilot cluster. Note that RAG requires the latest Autopilot features, available on the `RAPID` release channel.
+If you prefer to manage your own cluster, set `create_cluster = false` in the following section under `workloads.tfvars`. Creating a long-running cluster may be better for development, allowing you to iterate on Terraform components without recreating the cluster every time.
+
+Use gcloud to create a GKE Autopilot cluster. Note that RAG requires the latest Autopilot features, available on the `RAPID` release channel.
```

Comment:
Mention exactly which version is required?

Reply:
Actually this is a bit confusing - the section header says "use an existing cluster" but this paragraph mentions creating a new one.